_posts/2017-09-08-web.md

```diff
@@ -12,9 +12,84 @@ hierarchy:
+
+Therefore, many different servers hosts a small portion of the database, and they all use a common set of software to communicate between each other. We call these computers *DNS servers*. Since all of these DNS servers can communicate between one another, if a DNS server does not contain the domain name requested, it simply redirects it to the next server which may contain the desired information. This way, you're continually redirected down the hierachy until you get to the domain name you request (as in the figure below). Neat, huh?
+
+![Image](https://web.stanford.edu/class/msande91si/www-spr04/readings/week1/InternetWhitepaper_files/ruswp_diag6.gif)
+
+Diagram sourced from [stanford.edu](stanford.edu).
+
+> Check it out! What DNS server are you using? Type in `nslookup facebook.com` in either Command Prompt ([Windows][ins-w]), or Terminal ([Mac][ins-m]).
+
+##### HTTP and the World Wide Web
+
+So what really happens when you type in `google.com` into your browser? Once the Domain Name System translates the *domain name* into an *IP address*, your request is sent to the server to decode. This is done over a protocol called the **HyperText Transfer Protocol**, or **HTTP**. This is why most URLs are prefixed with `http://facebook.com`. When you add the `http://`, you're telling your web browser to use a specific protocol. 
+
+To provide some background, **hypertext** was called structured text that uses connections (links, or URLs) between servers containing text. Therefore, the *hypertext transfer protocol* specifies how to exchange and transfer this hypertext between servers. HTTP functions as a request-response protocol; that means that each response must be triggered by a corresponding request. To put this into practice, let's imagine our server requests a particular file, `silverman.jpg`. Our server could see this request, and will respond with a **status code** and a message payload. In this case, it will respond with *HTTP 200 OK*, meaning that the request was successful, and a copy of `silverman.jpg`. 
+ 
+It's important to understand that HTTP is a *connectionless*, text-based protocol. After the request is serviced, the connection between your web browser and a server is disconnected; a new connection must be made for each request. This is why Web Developers attempt to minimize *latency*, or the time to initialize a new connection, by bundling smaller requests for information together into one larger request. 
+
+##### Try it out! Simulate an HTTP request on your own.
+
+Now that you know how HTTP works, it might be useful to simulate how these transactions work on your own. In this exercise, we'll use telnet, a text-communication tool between servers, to simulate an HTTP request.
+
+> Note: if you're on Windows, you'll have to enable telnet using these [instructions](https://kb.ctera.com/article/how-to-open-a-telnet-session-on-windows-7-or-windows-8-os-16.html).
```

Comment:
Actually on Mac as well, `telnet` is not installed by default. I installed it via `brew install telnet`. This is probably the easiest way to instruct people.

You can link them to our [Mac setup instructions](https://starthacking.org/mac/), which has them install Homebrew, and say once you have that installed do `brew install telnet`.

Reply:
I think it came installed on linux, which is cool :open_mouth: